### PR TITLE
Intellij setup: Add back the Staging tests config file #7157

### LINKED ISSUE #7157: Intellij setup: Add back the Staging tests config file
The Intellij config file for staging tests was added in #6824 but has gone missing again.

### PR DESCRIPTION
**Pull request by LiHaoTan**
Fixes #7157 

**Outline of Solution**

- [x] Add run configurations for Eclipse and IntelliJ IDEA
- [x] Modify `TestNgTest.checkTestsInTestNg` as it uses a brittle approach of manually parsing XML

The test has to be modified because `testng-ci.xml` and `testng-staging.xml` now share a common TestNG suite and the current test is not able to parse child suites.

### CODE REVIEW

**Comment by LiHaoTan** [comment]
Ready for review

File: src/test/java/teammates/test/cases/testdriver/TestNgTest.java [hunk 1]
```diff
@@ -1,126 +1,130 @@
...
-                testFiles.putAll(
-                        addFilesToTestsRecursively(path + "/" + name,
-                                                   isPackageNameInTestNg(packageName + "." + name, testNgXml),
-                                                   packageName + "." + name, testNgXml));
-            }
+        final Configuration configuration = new Configuration();
+        SuiteRunner suiteRunner = new SuiteRunner(configuration, xmlSuite, TestNG.DEFAULT_OUTPUTDIR);
+        for (XmlTest test : xmlSuite.getTests()) {
+            TestRunner testRunner = new TestRunner(configuration, suiteRunner, test, false,
+                    null);
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/testdriver/TestNgTest.java]: This can be combined with the previous line?

**Comment by whipermr5** [comment]
I believe component tests need not be run again as they have already been run in CI. Staging tests are to test against the live server.

**Comment by LiHaoTan** [comment]
@whipermr5 yea it was my mistake, I blindly read the comment in `testng-ci.xml` and assumed that the only difference is `rare-tests`.

**Comment by LiHaoTan** [comment]
@whipermr5 I have a question though, I could not find any valid reason why `BackDoorTest` should not run during staging since it requires a server to run (and a local and staging server would not behave the same). I tried tracing through the [history](https://github.com/TEAMMATES/teammates/pull/5129/commits/231c9770f7620c74e9c2465efdb9a8aa1ce2131f#diff-09fbe34d8740a2d1485e2b38c6f38c04R55) and there is no context to it, just the comment.

**Comment by whipermr5** [comment]
@LiHaoTan I guess it's because the backdoor code usually does not change, so there is no need to run the test at every deployment.

**Comment by damithc** [comment]
> I have a question though, I could not find any valid reason why BackDoorTest should not run during staging since it requires a server to run (and a local and staging server would not behave the same

Most likely to reduce costs. Staging tests cost money. If backdoor test is broken, other tests are going to fail anyway. But if the test is not too heavy, I don't mind including it.

**Comment by whipermr5** [comment]
@LiHaoTan Let's finish this?

**Comment by LiHaoTan** [comment]
@whipermr5 sorry for the delay as I wanted to address some other things as well, will create a corresponding new issue soon to resolve some of the issues here.

**Comment by wkurniawan07** [comment]
I'll be closing this since the issue at hand is no longer relevant. I recognize that there are other fixes being done here, but they are significantly different from the original intention of the issue/PR. A new issue/PR will help to start fresh.

**Pull request closed without merging**